src/coreclr/jit/emitarm64.h

```diff
@@ -156,6 +157,9 @@ inline bool OptimizeLdrStr(instruction ins,
         return true;
     }
 
+    // Register 2 needs conversion to unencoded value for following optimisation checks.
+    reg2 = encodingZRtoSP(reg2);
```

Comment:
I didn't follow, why is this change needed? Was it missed when we implemented ldr -> ldp?

Reply:
never mind, just see that this is moved from `ReplaceLdrStrWithPairInstr`.